python/doc/examples/numerical_methods/optimization_bonmin.ipynb

```diff
@@ -0,0 +1,165 @@
+   "source": [
+    "from __future__ import print_function\n",
+    "import openturns as ot"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "B-BB\n",
+      "B-OA\n",
+      "B-QG\n",
+      "B-Hyb\n",
+      "B-iFP\n"
+     ]
+    }
+   ],
+   "source": [
+    "# List available algorithms\n",
+    "for algo in ot.Bonmin.GetAlgorithmNames():\n",
+    "    print(algo)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Details and references on bonmin algorithms are available [here](https://projects.coin-or.org/Bonmin)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setting up and solving a simple problem\n",
+    "\n",
+    "The following example will demonstrate the use of bonmin \"BB\" algorithm to solve the following problem:\n",
+    "\n",
+    "$$ \\min - x_0 - x_1 - x_2 $$\n",
+    "    \n",
+    "such that:\n",
+    "    \n",
+    "$$\\begin{array}{l}\n",
+    "(x_1 - \\frac{1}{2})^2 + (x_2 - \\frac{1}{2})^2 \\leq \\frac{1}{4} \\\\\n",
+    "x_0 - x_1 \\leq 0 \\\\\n",
+    "x_0 + x_2 + x_3 \\leq 2\\\\\n",
+    "x_0 \\in \\{0,1\\}^n\\\\\n",
+    "(x_1, x_2) \\in \\mathbb{R}^2\\\\\n",
+    "x_3 \\in \\mathbb{N}\n",
+    "\\end{array} $$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Definition of objective function\n",
+    "objectiveFunction = ot.SymbolicFunction(['x0','x1','x2','x3'], ['-x0 -x1 -x2'])\n",
+    "\n",
+    "# Definition of variables bounds\n",
+    "bounds = ot.Interval([0,0,0,0],[1,1e99,1e99,5],[True,True,True,True],[True,False,False,True])\n",
+    "\n",
+    "# Definition of constraints\n",
+    "# Constraints in OpenTURNS are defined as g(x) = 0 and h(x) >= 0\n",
+    "#    No equality constraint -> nothing to do\n",
+    "#    Inequality constraints:\n",
+    "h = ot.SymbolicFunction(['x0','x1','x2','x3'], ['-(x1-0.5)^2 - (x2-0.5)^2 + 0.25', 'x1 - x0', '-x0 - x2 - x3 + 2'])\n",
+    "    \n",
+    "# Definition of variables types\n",
+    "variablesType = [ot.OptimizationProblemImplementation.BINARY,ot.OptimizationProblemImplementation.CONTINUOUS,ot.OptimizationProblemImplementation.CONTINUOUS,ot.OptimizationProblemImplementation.INTEGER]\n",
+    "\n",
+    "# Setting up Bonmin problem\n",
+    "problem = ot.OptimizationProblem(objectiveFunction)\n",
+    "problem.setBounds(bounds)\n",
+    "problem.setVariablesType(variablesType)\n",
+    "problem.setInequalityConstraint(h)\n",
+    "\n",
+    "bonminAlgorithm = ot.Bonmin(problem,'B-BB')\n",
+    "bonminAlgorithm.setMaximumEvaluationNumber(10000)\n",
+    "bonminAlgorithm.setMaximumIterationNumber(1000)\n",
+    "bonminAlgorithm.setStartingPoint([0,0,0,0])\n",
+    "\n",
+    "ot.ResourceMap.SetAsString('Bonmin-mu_oracle','loqo')\n",
```

Comment:
Please explain why these statements are useful in this specific example.

Reply:
Actually they're not :-)
The problem would be solved just fine without this line and the next one. The point here is just to demonstrate the use of ResourceMap to set Bonmin and Ipopt options.

Reply:
If these lines are not useful, I suggest to 
* remove them in this specific example,
* create another example in which this configuration is appropriate.